sycl/test/host-interop-task/interop-task.cpp

```diff
@@ -0,0 +1,187 @@
+void test1() {
+  static constexpr int COUNT = 4;
+  queue Q;
+  buffer<int, 1> Buffer1{BUFFER_SIZE};
+  buffer<int, 1> Buffer2{BUFFER_SIZE};
+
+  // init the buffer with a'priori invalid data
+  init<int, -1, -2>(Buffer1, Buffer2, Q);
+
+  // Repeat a couple of times
+  for (size_t Idx = 0; Idx < COUNT; ++Idx) {
+    copy(Buffer1, Buffer2, Q);
+    modify(Buffer2, Q);
+    copy(Buffer2, Buffer1, Q);
+  }
+
+  {
+    auto Acc = Buffer1.get_access<mode::read>();
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx) {
+      std::cout << "First buffer [" << Idx << "] = " << Acc[Idx] << std::endl;
+      assert((Acc[Idx] == COUNT - 1) && "Invalid data in the first buffer");
+    }
+  }
+  {
+    auto Acc = Buffer2.get_access<mode::read>();
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx) {
+      std::cout << "Second buffer [" << Idx << "] = " << Acc[Idx] << std::endl;
+      assert((Acc[Idx] == COUNT - 1) && "Invalid data in the second buffer");
+    }
+  }
+}
+
+// Same as above, but performing each command group on a separate SYCL queue
+// (on the same or different devices). This ensures the dependency tracking
+// works well but also there is no accidental side effects on other queues.
+void test2() {
+  static constexpr int COUNT = 4;
+  buffer<int, 1> Buffer1{BUFFER_SIZE};
+  buffer<int, 1> Buffer2{BUFFER_SIZE};
+
+  // init the buffer with a'priori invalid data
+  {
+    queue Q;
+    init<int, -1, -2>(Buffer1, Buffer2, Q);
+  }
+
+  // Repeat a couple of times
+  for (size_t Idx = 0; Idx < COUNT; ++Idx) {
+    queue Q;
+    copy(Buffer1, Buffer2, Q);
+    modify(Buffer2, Q);
+    copy(Buffer2, Buffer1, Q);
+  }
+
+  {
+    auto Acc = Buffer1.get_access<mode::read>();
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx) {
+      std::cout << "First buffer [" << Idx << "] = " << Acc[Idx] << std::endl;
+      assert((Acc[Idx] == COUNT - 1) && "Invalid data in the first buffer");
+    }
+  }
+  {
+    auto Acc = Buffer2.get_access<mode::read>();
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx) {
+      std::cout << "Second buffer [" << Idx << "] = " << Acc[Idx] << std::endl;
+      assert((Acc[Idx] == COUNT - 1) && "Invalid data in the second buffer");
+    }
+  }
+}
+
+// A test that does a clEnqueueWait inside the interop scope, for an event
+// captured outside the command group. The OpenCl event can be set after the
+// command group finishes. Must not deadlock according to implementation and
+// proposal
+void test3() {
+  // Want some large buffer for operation to take long
+  buffer<int, 1> Buffer{BUFFER_SIZE * 128};
+
+  queue Q;
+
+  event Event = Q.submit([&](handler &CGH) {
+    auto Acc1 = Buffer.get_access<mode::write>(CGH);
+
+    CGH.parallel_for<class Init3>(BUFFER_SIZE, [=](item<1> Id) {
+      Acc1[Id] = 123;
+    });
+  });
+
+  Q.submit([&](handler &CGH) {
+    CGH.codeplay_host_task([=](interop_handle IH) {
+      cl_event Ev = Event.get();
+
+      int RC = clWaitForEvents(1, &Ev);
```

Comment:
Please, add OpenCL requirement at the beginning of the test:
```c++
// REQUIRES: opencl
```

Reply:
Done.